File: Source/Bifrost.JavaScript.Specs/utils/for_StringMapping/when_getting_values_from_placeholders.js
describe("when getting values from placesholders", function () {
    var expectedResult = "Say/hello/to/mr.potatohead";
    var mapping = Bifrost.StringMapping.create({
        format: "{something}/{else}",
        mappedFormat: "Say/{else}/to/{something}"
    });
    var result = mapping.getValues("mr.potatohead/hello");

    it("should contain the something placeholder", function () {
        expect(result.something).toBe("hello");
    });
});

File: Source/Bifrost.Specs/Utils/for_StringMapping/when_getting_values_from_placeholders.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Bifrost.Utils;
using Machine.Specifications;

namespace Bifrost.Specs.Utils.for_StringMapping
{
    public class when_getting_values_from_placeholders
    {
        const string expected_result = "Say/hello/to/mr.potatohead";

        static StringMapping mapping = new StringMapping(
                "{something}/{else}",
                "Say/{else}/to/{something}"
            );

        static dynamic result;

        Because of = () => result = mapping.GetValues("mr.potatohead/hello");

        It should_contain_the_something_placeholder = () => ShouldExtensionMethods.ShouldEqual(result.something, "hello");
    }
}
